python/taichi/lang/meta.py

```diff
@@ -21,6 +21,16 @@ def cook_image_type(x):
     return x
 
 
+@ti.kernel
+def vector_to_fast_image(img: ti.template(), out: ti.ext_arr()):
+    # FIXME: Why is ``for i, j in img:`` slower than:
+    for i, j in ti.ndrange(*img.shape):
+        u, v, w = min(255, max(0, int(img[i, img.shape[1] - 1 - j] * 255)))
+        # We use i32 for |out| since OpenGL and Metal doesn't support u8 types
+        # TODO: treat Cocoa and Big-endian machines, with XOR logic
+        out[j * 1920 + i] = w + (v << 8) + (u << 16)
```

Comment:
`img.shape[0]` actually.
Btw, I saw you use RGB order in cocoa.cpp, and BGR order in x11.cpp and win32.cpp. I want to know if mac actually use big-endian? If so, no changes need to be done here. Otherwise `ti.static(ti.get_os_name() == 'osx')` would be used.